From 482c573d9e87d15632ac16941d03684270854d30 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Wed, 30 Nov 2011 21:48:26 +0000
Subject: [PATCH] Only install files from /etc/ld.so.conf.d/ directory

When calling inst_simple() it will ignore anything that
is not a file, so harden the check done before the calling.
---
 dracut |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/dracut b/dracut
index 8449fc1..56c8df3 100755
--- a/dracut
+++ b/dracut
@@ -712,7 +712,7 @@ done
 if [[ $kernel_only != yes ]]; then
     # make sure that library links are correct and up to date
     for f in /etc/ld.so.conf /etc/ld.so.conf.d/*; do
-        [[ -e $f ]] && inst_simple "$f"
+        [[ -f $f ]] && inst_simple "$f"
     done
     if ! ldconfig -r "$initdir"; then
         if [[ $UID = 0 ]]; then
